app/controllers/v0/sessions_controller.rb

```diff
@@ -3,20 +3,51 @@
 
 module V0
   class SessionsController < ApplicationController
-    skip_before_action :authenticate, only: [:new, :saml_callback, :saml_logout_callback]
+    skip_before_action :authenticate, only: [:new, :authn_urls, :saml_callback, :saml_logout_callback]
 
     STATSD_LOGIN_FAILED_KEY = 'api.auth.login_callback.failed'
     STATSD_LOGIN_TOTAL_KEY  = 'api.auth.login_callback.total'
 
+    # Collection Action: this method will eventually be replaced by auth_urls
     def new
       saml_auth_request = OneLogin::RubySaml::Authrequest.new
+      authn_context = LOA::MAPPING.invert[params[:level]&.to_i] || LOA::MAPPING.invert[1]
+      saml_settings = saml_settings(authn_context: authn_context)
       render json: { authenticate_via_get: saml_auth_request.create(saml_settings, saml_options) }
     end
 
+    # Collection Action: method will eventually replace new
+    # no auth required
+    def authn_urls
+      render json: {
+        mhv: build_url(authn_context: 'mhv', connect: 'mhv'),
```

Comment:
I notice the existing values for authn_context (taken from `LOA::MAPPING` are url-like identifiers:
```
{1=>"http://idmanagement.gov/ns/assurance/loa/1", 2=>"http://idmanagement.gov/ns/assurance/loa/2", 3=>"http://idmanagement.gov/ns/assurance/loa/3"}
```
Are the values meaningful, or it just an opaque string that we need to send and get back in the response?

Reply:
The latter.  ID.me instructed that we place these in `<AuthNRequest>` and that we we will receive them back in `<AuthNResponse>`.  Might as well bee `foo1` and `foo3` respectively.

Reply:
they're meaningful to id.me, I suppose they could have done the local mapping on their end we would just pass something like 'idmeloa1' 'idmeloa3' instead, but that's how it is if we're to be backward compatible.

Reply:
@omgitsbillryan FWIW they lied about passing them back. We don't get the authn_request back in the saml_response based on my findings, hence why I had to do that stupid heuristic in /saml/user to determine the authn_context.

Reply:
I see an AuthnContext element in the saml response using idme-style (i.e. current master) login

Reply:
It's in the raw XML, I don't know if ruby-saml exposes it.

Reply:
ruby-saml doesn't expose it, which is why there's the REXML parsing in the idme user_attributes class. AuthnContext is exactly the element it's using to derive the loa.

Reply:
Would you prefer I parse the xml to get it or are you satisfied with my use of the dslogon? and mhv? helper methods to determine the authn_context based on the attributes available? I don't have strong opinion either way.